crates/reward/src/lib.rs

```diff
@@ -182,12 +186,12 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
             .ok_or(Error::<T, I>::TryIntoIntError)?)
     }
 
-    pub fn deposit_stake(vault_id: &DefaultVaultId<T, I>, amount: SignedFixedPoint<T, I>) -> Result<(), DispatchError> {
-        checked_add_mut!(Stake<T, I>, vault_id, &amount);
+    pub fn deposit_stake(reward_id: &T::RewardId, amount: SignedFixedPoint<T, I>) -> Result<(), DispatchError> {
+        checked_add_mut!(Stake<T, I>, reward_id, &amount);
         checked_add_mut!(TotalStake<T, I>, &amount);
 
-        for currency_id in [vault_id.wrapped_currency(), T::GetNativeCurrencyId::get()] {
-            <RewardTally<T, I>>::mutate(currency_id, vault_id, |reward_tally| {
+        for currency_id in [T::GetNativeCurrencyId::get(), T::GetWrappedCurrencyId::get()] {
```

Comment:
Why do you revert back to `GetWrappedCurrencyId` instead of using the vault_id?

Reply:
Because the `vault_id` is now the associated type `RewardId`.

Reply:
Why did you introduce the `RewardId` though? I mean, is there a concrete reason for this, or is it just a code style preference?

Reply:
Ah sorry yep I should've made it clear, it's because the stake-to-vote rewards are for normal accounts so I needed some way to separate the two.